### PR TITLE
added resize image avatar functionality

### PR DESCRIPTION
**Pull request by greentoss**
## Description

added functionality of auto resize avaar image
it looks like dimension 200x200 is more useful then 100x100
still need to increase tests coverage

### Checklist

- [ ] 🔽 My branch is up-to-date with "development" branch
- [ ] ✅All tests passed locally
- [ ] ✨My changes working with up-to-date admin and back-end part locally, like charm
- [ ] 🔗 Link pull request to issue

### CODE REVIEW

File: src/pages/profile-page/avatar/avatar.js [hunk 1]
```diff
@@ -33,42 +32,57 @@ const Avatar = ({ setUserImageUrl, userImageUrl, setUpload, setDeleteAvatar, t }
...
-  const checkDimension = async (result) => {
-    const image = new Image();
-
-    const promiseImg = new Promise((resolve, reject) => {
-      image.onload = () => {
-        if (image.height <= 100 && image.width <= 100) {
-          resolve(true);
-        } else {
-          resolve(false);
+  const checkOrResizeImageTo100x100 = (base64Str, maxWidth = 200, maxHeight = 200) => new Promise((resolve) => {
```
Comment (maarekua) [inline src/pages/profile-page/avatar/avatar.js:35]: General functionality that you can reuse should be moved to utils folders.

File: src/pages/profile-page/avatar/avatar.js [hunk 2]
```diff
@@ -33,42 +32,57 @@ const Avatar = ({ setUserImageUrl, userImageUrl, setUpload, setDeleteAvatar, t }
...
+        height = maxHeight;
+      }
+      canvas.width = width;
+      canvas.height = height;
+      canvasCtx.drawImage(originalImage, 0, 0, width, height);
+      resolve(canvas.toDataURL('image/jpeg', 1));
+    };
+  });
+
+  const dataURItoBlob = (dataURI) => {
```
Comment (maarekua) [inline src/pages/profile-page/avatar/avatar.js:59]: Same here.

File: src/pages/profile-page/avatar/avatar.js [hunk 3]
```diff
@@ -82,17 +96,17 @@ const Avatar = ({ setUserImageUrl, userImageUrl, setUpload, setDeleteAvatar, t }
 
   return (
     <div className={classes.imageContainer}>
-      {userImageUrl && (
+      {userImageUrl ? (
```
Comment (maarekua) [inline src/pages/profile-page/avatar/avatar.js:99]: Is there any reason to change it to ternary?

File: src/tests/unit/pages/profile-page/avatar/avatar.spec.js [hunk 4]
```diff
@@ -44,25 +44,26 @@ describe('Profile=Page test image restrictions', () => {
...
+    const file = new File(['test file'], 'image.jpeg', { type: 'image/png' });
 
     fireEvent.change(uploader, { target: { files: [file] } });
 
-    expect(await screen.findByText('error.profile.dimension')).toBeInTheDocument();
+    expect(uploader.type).toBe('file');
+
+    expect(uploader.files.length).toBe(1);
   });
+  // it('should show preview image', async () => {
```
Comment (maarekua) [inline src/tests/unit/pages/profile-page/avatar/avatar.spec.js:56]: Add explanation to a commented piece of code, or remove it.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/pages/profile-page/avatar/avatar.js (hunk 1 above)
Comment (greentoss) [reply src/pages/profile-page/avatar/avatar.js]: general functions have been moved to utils folder.

File: src/pages/profile-page/avatar/avatar.js (hunk 2 above)
Comment (greentoss) [reply src/pages/profile-page/avatar/avatar.js]: general functions have been moved to utils folder.

File: src/pages/profile-page/avatar/avatar.js (hunk 3 above)
Comment (greentoss) [reply src/pages/profile-page/avatar/avatar.js]: returned to previous state

File: src/tests/unit/pages/profile-page/avatar/avatar.spec.js (hunk 4 above)
Comment (greentoss) [reply src/tests/unit/pages/profile-page/avatar/avatar.spec.js]: tests have been refactored, comments deleted

**Review by maarekua — APPROVED** [review]

File: src/utils/checkOrResizeImage.spec.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+describe('checkOrResizeImage test', () => {
+  const canvas = document.createElement('canvas');
+
```
Comment (greentoss) [inline src/utils/checkOrResizeImage.spec.js:8]: this function  **checkOrResizeImage**  draws canvas
as to google search - the tests for this function requires  **jest-canvas-mock** dev dependancy
can i use it or how should i test instead ?

**Review by maarekua — APPROVED** [review]

File: src/utils/checkOrResizeImage.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  setUserImageUrl,
+  setUpload,
+  maxWidth = 200,
+  maxHeight = 200
+) => {
+  const canvas = document.createElement('canvas');
+  const canvasCtx = canvas.getContext('2d');
+  let width = originalImage.naturalWidth;
+  let height = originalImage.naturalHeight;
+  if (width >= height) {
```
Comment (maarekua) [inline src/utils/checkOrResizeImage.js:13]: Can you refactor this part? Thanks.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/utils/checkOrResizeImage.js (hunk 6 above)
Comment (greentoss) [reply src/utils/checkOrResizeImage.js]: changed to nested if statement to  && operator

**Review by maarekua — APPROVED** [review]

**Review by MickhaelSh — APPROVED** [review]

**Pull request merged**